dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java

```diff
@@ -78,6 +75,8 @@ ProcessInstance handleCommand(String host,
 
     ProcessDefinition findProcessDefinition(Long processDefinitionCode, int processDefinitionVersion);
 
+    List<ProcessDefinition> findProcessDefinitions(Map<Long, Integer> codeVersionMap);
```

Comment:
```suggestion
    List<ProcessDefinition> findProcessDefinitionsByProcessInstances(List<ProcessInstance> processInstances);
```

Reply:
I found that there is no `ProcessDefinitionLogService` in module `dolphinscheduler-service`.Do you mean that i need to create a new class in `dolphinscheduler-service` under package `org.apache.dolphinscheduler.service.process`?

Reply:
Ok, I see the latest code, there is no `ProcessDefinitionLogService`, you can add `findProcessDefinitionsByProcessInstances(List<ProcessInstance> processInstances)` in `ProcessDefinitionDaoImpl` class.

Reply:
But how can i use this method in `MasterFailoverService`? Inject `ProcessDefinitionDaoImpl `class directly ? Is that suitable for Dolphin's convention?

Reply:
OK， I see the `DAO` layer packages the `Mapper` layer, it's kinda suitable for this purpose👌